frontend/index.ts

```diff
@@ -1,13 +1,20 @@
 /*global moment*/
+import moment from "moment";
+
+class Exam {
+    public start: moment.Moment;
+    public end: moment.Moment;
+    public past: boolean;
+}
 
 window.onload = function () {
-    const times = [];
+    const times: Exam[] = [];
     const now = moment();
 
     // Template creates divs with class "final" for each final
     // Filter out finals with dates in the past, and add the class "past",
     //  which will apply a style to hide the element
-    const finalCards = Array.from(document.getElementsByClassName("final")).filter((f) => {
+    const finalCards = Array.from(document.querySelectorAll(".final")).filter((f: HTMLDivElement) => {
```

Comment:
Per [this StackOverflow answer](https://stackoverflow.com/a/43967953/5991492), `getElementsByClassName` is literally 100x faster than `querySelectorAll`